### PR TITLE
push top_k negations down into reducer

### PR DESCRIPTION
**Pull request by bosconi**
Push negation in `build_topk_negated_stage` down into reducer.

### Motivation

This PR refactors existing code. The negation was being performed at a higher level, so that more work was done than necessary.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
- [ ] 
  -

### CODE REVIEW

**Comment by bosconi** [comment]
@antiguru, as discussed. Should the comments for the changed lines of code be revised to be more clear?

**Review by antiguru — APPROVED** [review]
Looks good, but we should update the documentation in the code. The function `build_topk_negated_stage` now has some comments, inline and in the function documentation, which need to be flipped.

File: src/compute/src/render/top_k.rs [hunk 1]
```diff
@@ -439,8 +439,8 @@ where
         };
         // Turn input into collection.
         let input = input.as_collection(|k, v| (k.into_owned(), v.into_owned()));
-        // Negate oks and concatenate them with the input.
-        (oks.negate().concat(&input), errs)
+        // Concatenate oks with the input.
```
Comment (teskje) [inline src/compute/src/render/top_k.rs:442]: Nit and not on you, so feel free to ignore, but this comment just re-states the next line of very obvious code without adding any additional insight, so it seems unnecessary. Same for the "Turn input into collection." one above.

Comment (bosconi) [reply src/compute/src/render/top_k.rs:442]: Agreed. Removing both comments.

**Pull request merged**